src/utils/relative_point.rs

```diff
@@ -0,0 +1,31 @@
+#[cfg(feature = "wayland_frontend")]
+use crate::wayland::seat::WaylandFocus;
+
+use super::{Logical, Point};
+
+/// Represents a point on a surface, relative to that surface's origin in
+/// global compositor space.
+#[derive(Debug, Clone)]
+pub struct RelativePoint<S> {
```

Comment:
That's what I originally had, but I realized I had to move it out of the `input::pointer` mod because it's used by `grab` code. Once I moved it into `utils`, it seemed appropriate to give it more of a general name (`Focus` being an input-specific concept).

Happy to place/name it whatever you like, just wanted to add that context. My suggestion given your constraints would be `SurfaceCoordinate` or `SurfacePoint` or similar.

Reply:
> SurfacePoint

That is also a good one, but given a focused element doesn't have to be a surface, `RelativePoint` does seem to capture this quite well. 😅